themes/default/content/blog/benchmarking-python-performance/index.md

```diff
@@ -0,0 +1,216 @@
+---
+title: "Benchmarking Python Performance"
+date: 2023-07-31
+meta_desc: "Benchmarking and improving the performance of Pulumi Python programs."
+meta_image: meta.png
+
+authors:
+    - robbie-mckinstry
+
+tags:
+    - performance
+    - platform
+    - engineering
+---
+
+This is the second post in a series about performance optimizations we've made
+to the Pulumi CLI and SDKs. In this post, we'll go deep on a performance
+improvement we made for Pulumi Python programs. You can read more
+about Amazing Performance in
+[the first post in the series](https://www.pulumi.com/blog/amazing-performance/).
+
+<!--more-->
+
+We recently took a hard look at the performance of Python programs when we
+realized they weren't performing up to our expectations. We uncovered a major
+bug limiting Python performance, and we ran a number of rigorous experiments
+to evaluate just how performant Pulumi Python programs are after the bug had
+been repaired. The results indicate Pulumi Python programs are significantly
+faster than they were, and now Pulumi Python has reached performance parity
+with Pulumi Node!
+
+## The Bug
+
+When you execute a Pulumi program, Pulumi internally builds a dependency graph
+between the resources in your program. In every Pulumi program, some resources
+have all their input arguments available at the time of their construction.
+In contrast, other resources may depend on `Outputs` from other resources.
+
+For example, consider a sample program where we expose a compute instance
+publicly using a floating IP:
+
+```javascript
```

Comment:
Since the article is about Python improvements, could we rewrite this example in Python?